### ISSUE TITLE
About multiple layer menu

### ISSUE THREAD

**Issue by MatheusNCarvalho** [creativetimofficial/vue-paper-dashboard-pro#17]
Hey, i'm using your template to make a big application front-end, and i'm really liking it, but i will need multiple layers in the menu(like child-layers with child-layers), is there a way i can achieve that? or someway you could help? i've seen on the template site that you have some work on it, and plan to add this feature in next updates.

I'm waiting your reply, Thanks man.

**Comment by cristijora** [comment]
Hi @MatheusNCarvalho Glad you like the template and find it useful.
Indeed, the current version does not support multiple menus. I have a started version of a sidebar plugin which supports multiple menus. https://github.com/cristijora/vue-side-bar
You can achieve this by wrapping the links into a more generic component similar to [this one](https://github.com/cristijora/vue-side-bar/blob/master/src/components/SidebarItem.vue)


I will post a working example for the current paper-dashboard
1. Delete everything under `ul :class="navClasses"` in `Sidebar.vue` so you are left with something like this: 
```html
<div class="sidebar-wrapper" ref="sidebarScrollArea">
      <slot>

      </slot>
      <ul :class="navClasses" >
      </ul>
    </div>
```
2. Create a new component called **SidebarItem.vue**
```html
<template>
  <li>
    <a v-if="link.children" data-toggle="collapse" href="javascript:void(0)"
       @click="collapseMenu(index)">
      <i :class="link.icon"></i>
      <p>{{link.name}}
        <b class="caret"></b>
      </p>
    </a>

    <div v-if="link.children"
         aria-expanded="true">
      <el-collapse-transition>
        <ul class="nav" v-show="!link.collapsed">
          <template v-for="(subLink, subLinkIndex) in link.children">
            <sidebar-item v-if="subLink.children && subLink.children.length > 0"
                          :link="subLink"
                          :index="index"
                          :class="{active: subLink.active}"
                          @collapse-menu="collapseSubMenu(subLink)">
            </sidebar-item>
            <component v-else
              :is="elementType(subLink)"
              :key="subLink.path"
              :to="subLink.path" tag="li"
              :class="{active: subLink.active}"
              :ref="subLink.name">
              <a :href="elementType(subLink) === 'li' ? subLink.path : '#'" :target="subLink.target">
                <span class="sidebar-mini">{{linkAbbreviation(subLink.name)}}</span>
                <span class="sidebar-normal">{{subLink.name}}</span>
              </a>
            </component>

          </template>

        </ul>
      </el-collapse-transition>
    </div>


    <component v-else
               :to="link.path"
               :is="elementType(link, false)"
               :class="{active: link.active}"
               :target="link.target"
               :href="link.path">
      <i :class="link.icon"></i>
      <p>{{link.name}}</p>
    </component>

  </li>
</template>
<script>
  import CollapseTransition from 'element-ui/lib/transitions/collapse-transition'
  export default {
    components: {
      [CollapseTransition.name]: CollapseTransition
    },
    props: {
      link: {
        type: Object,
        default: () => {
          return {
            name: '',
            path: '',
            children: []
          }
        }
      },
      index: Number
    },
    methods: {
      elementType (link, isParent = true) {
        if (link.isRoute === false) {
          return isParent ? 'li' : 'a'
        } else {
          return 'router-link'
        }
      },
      linkAbbreviation (name) {
        const matches = name.match(/\b(\w)/g)
        return matches.join('')
      },
      collapseMenu (index) {
        this.$emit('collapse-menu', index)
      },
      collapseSubMenu (link) {
        link.collapsed = !link.collapsed
      }
    }
  }
</script>
<style>
</style>
```
3. Go to `SidebarPlugin/index.js` and import the new component
```js
import SidebarItem from './SidebarItem.vue'
//inside install function
install (Vue, options){
 // other code
Vue.component('sidebar-item', SidebarItem)
}
```

4. Add the component inside Sidebar.vue
```html
 <ul :class="navClasses" >
        <sidebar-item v-for="(link, index) in sidebarLinks"
                      :link="link"
                      :index="index"
                      :class="{active: isMenuActive(link) || link.active}"
                      @collapse-menu="collapseMenu">
        </sidebar-item>
</ul>
```

5. Feel free to add children to one of the links inside a menu
Example
```js
{
    name: 'Dashboard',
    icon: 'ti-panel',
    collapsed: true,
    children: [{
      name: 'Overview',
      icon: 'ti-panel',
      path: '/admin/overview',
      collapsed: true,
      children: [{
        name: 'Overview',
        path: '/components/grid-system'
      },
      {
        name: 'Stats',
        path: '/components/panels'
      }]
    },
    {
      name: 'Stats',
      path: '/admin/stats'
    }]
  },
```
Make sure that your menu has `children` array and `collapsed` property set initially.


Notes: You might need to style a bit the sub-menus so their sub-links have more padding or smth like that.
Haven't tested if the menu inside menu gets activated whenever one of its children are active.

**Comment by JuanesAdriano** [comment]
Thanks for the support man! worked fine for me, but i still have one question, where exactly can i add style to the sub-menus (for add more padding) without messing with the other menu Styles?

**Comment by cristijora** [comment]
@JuanesAdriano I think it would be best to add them in a separate scss file on inside `Sidebar.vue` itself in case you want make use of further updates from us like minor fixes. Ideally the current scss files should be left as they are so you can replace them later with ones from a newer version.

**Comment by JuanesAdriano** [comment]
![image](https://user-images.githubusercontent.com/11681859/31137145-b46a7044-a840-11e7-84cd-734dcb400d0e.png)

Here?

**Comment by cristijora** [comment]
Either `Sidebar.vue` inside `<style>` tag 
or create  a new file for example 
`_sidebar-menu.scss`
and import it inside `paper-dashboard.css`

**Comment by JuanesAdriano** [comment]
Got it, where i should apply this class now? to apply the padding just for the lower menu links?(Item>subItem>subsubitem)

**Comment by cristijora** [comment]
This has been fixed and released in v1.1.0
Feel free to download the [New version here](https://www.creative-tim.com/product/vue-paper-dashboard-pro) and follow along the [Migration tips](https://cristijora.github.io/paper-dashboard-pro/documentation.html?selectedKind=General%20tips&selectedStory=Migration%20guide&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel) from previous version to the new version.

The sidebar was improved and simplified a lot overall. It now relies on the same logic with sidebar-items from above.
